### PR TITLE
API Server: Use versioned objects for GET and CONNECT operations

### PR DESCRIPTION
**Pull request by csrwng**
Currently, parameters for GET or CONNECT operations are not documented in the swagger spec. This fixes that. The API installer needed to use the versioned set of options and not the internal ones.

### CODE REVIEW

**Comment by csrwng** [comment]
@lavalamp @smarterclayton ptal

**Comment by smarterclayton** [comment]
LGTM but we need a test.

**Comment by smarterclayton** [comment]
ok to test

**Comment by lavalamp** [comment]
LGTM once tested + working

**Comment by ixdy** [comment]
ran out of network quota, sorry about that.

@k8s-bot test this please

**Comment by lavalamp** [comment]
Can you please add a test? Also, shouldn't there be an update to the swagger stuff?

**Comment by lavalamp** [comment]
(That is why shippable failed: ```/root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/api/swagger-spec is out of date. Please run hack/update-swagger-spec.sh```)

**Comment by csrwng** [comment]
I will update it when I get back from vacation next week

**Comment by lavalamp** [comment]
@csrwng we're taking PRs again, hope you're back from vacation

@nikhiljindal can you review this one?

**Comment by nikhiljindal** [comment]
Sure. @csrwng Let me know when you update the PR

**Comment by csrwng** [comment]
@nikhiljindal - updated the PR with generated swagger doc and test cases

File: api/swagger-spec/v1.json [hunk 1]
```diff
@@ -5707,6 +5707,54 @@
       "nickname": "connectGetNamespacedPodExec",
       "parameters": [
        {
+        "type": "boolean",
+        "paramType": "query",
+        "name": "stdin",
+        "description": "redirect the standard input stream of the pod for this call; defaults to false",
```
Comment (nikhiljindal) [inline api/swagger-spec/v1.json]: redirect it where?

Comment (csrwng) [inline api/swagger-spec/v1.json]: I believe it means that stdin is redirected into the process you are executing in the pod.
However, this PR is not introducing a new description for this (maybe should be clarified in a separate PR). This PR simply exposes these parameters and descriptions that were already in the type annotations. 
`api/swagger-spec/v1.json` is a generated file.

File: pkg/apiserver/api_installer.go [hunk 2]
```diff
@@ -204,10 +204,11 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
...
-		getSubpath     bool
-		getSubpathKey  string
+		getOptions          runtime.Object
+		versionedGetOptions runtime.Object
+		getOptionsKind      string
+		getSubpath          bool
+		getSubpathKey       string
 	)
 	if isGetterWithOptions {
 		getOptions, getSubpath, getSubpathKey = getterWithOptions.NewGetOptions()
```
Comment (nikhiljindal) [inline pkg/apiserver/api_installer.go]: Sorry I cant find where is this NewGetOptions() method implemented?
https://github.com/GoogleCloudPlatform/kubernetes/search?utf8=%E2%9C%93&q=NewGetOptions returns the implementation for PodLogOptions only.
I am wondering why does getOptions not have any of these parameters?

File: api/swagger-spec/v1.json (hunk 1 above)
Comment (nikhiljindal) [inline api/swagger-spec/v1.json]: Yes. Seems like those descriptions were not used anywhere before this PR. Will appreciate if you can fix them. Not necessarily required in this PR.

**Comment by nikhiljindal** [comment]
LGTM, thanks!

File: pkg/apiserver/api_installer.go (hunk 2 above)
Comment (csrwng) [inline pkg/apiserver/api_installer.go]: Yes, it's currently only implemented by the pod's LogREST storage. It's purpose is to allow the implementing REST storage to receive additional parameters in its get request. The getOptions object is meant to be introspected to correlate request parameters to options object fields, the subpath and subpathkey make it possible to set one of the fields to correspond to the path parameter in the request url.
